### PR TITLE
moved support for arrays into whenArray function. when now behaves just like $.when

### PR DESCRIPTION
**Pull request by antialias**

### CODE REVIEW

File: deferred.coffee [hunk 1]
```diff
@@ -135,9 +135,9 @@ Deferred = ->
   return this
 
 # If we're dealing with multiple deferreds, it would be nifty to have a way to run code after all of them succeed (or any of them fail).
-# Let's set up a `.when([deferreds])` method to do that. It should be able to take any number or deferreds as arguments (or an array of them).
+# Let's set up a `.when([deferreds])` method to do that. It should be able to take any number or deferreds as arguments.
```
Comment (antialias) [inline deferred.coffee]: I made it shorter and this is the thanks I get!

**Comment by antialias** [comment]
js should be rebuild after merge

**Comment by antialias** [comment]
bump

**Comment by antialias** [comment]
Is this project still alive?

**Comment by antialias** [comment]
bump

**Comment by antialias** [comment]
bump

**Comment by antialias** [comment]
bump

**Comment by antialias** [comment]
bump

**Comment by sudhirj** [comment]
Sorry, I completely zoned out on Github notifications. If you're still interested, could you help me out a bit here? Why do we need a separate method for this, doesn't the `when` method accept an array as well? Or are you saying the original jQuery method doesn't?

**Comment by antialias** [comment]
> doesn't the when method accept an array as well?

[`$.when`](https://api.jquery.com/jquery.when/) does not work with arrays when called directly. The usual pattern for passing it an array is `$.when.apply(null, [foo, bar,baz])`.

**Comment by sudhirj** [comment]
Got it... I'd rather not add a method that isn't present in jQuery - and even jQuery is started with Promises compatibility now, so there's really no reason to diverge off the API. 

If there's a deviation from the jQuery API behaviour let's try to fix that instead. If not, the API on this library is effectively frozen - users should probably move to promises instead anyway.

**Comment by antialias** [comment]
Ok, I removed the new `whenArray` function. This PR still removes array support from `when`, which restores parity with `jQuery.when`.

**Comment by antialias** [comment]
bump @sudhirj

**Comment by sudhirj** [comment]
@antialias Released... sorry for keeping you waiting so long :(

**Pull request merged**